x/erc20/keeper/ibc_callbacks.go

```diff
@@ -101,15 +100,25 @@ func (k Keeper) OnRecvPacket(
 		return ack
 	}
 
-	pair, _ := k.GetTokenPair(ctx, pairID)
-	if !pair.Enabled {
+	pair, found := k.GetTokenPair(ctx, pairID)
+	if !(found && pair.Enabled) {
 		// no-op: continue with the rest of the stack without conversion
 		return ack
 	}
 
+	address := common.HexToAddress(pair.Erc20Address)
+	if !k.IsERC20Registered(ctx, address) {
+		err := errorsmod.Wrapf(types.ErrERC20TokenPairNotRegistered, "erc20 token pair %s does not exist", coin.Denom)
+		return channeltypes.NewErrorAcknowledgement(err)
+	}
+
 	// Instead of converting just the received coins, convert the whole user balance
 	// which includes the received coins.
 	balance := k.bankKeeper.GetBalance(ctx, recipient, coin.Denom)
+	if !balance.IsPositive() {
+		// if the balance is zero, no need to convert
+		return ack
```

Comment:
why not letting the convert flow run and fail? why explicitly check for balance here